pkg/sql/conn_executor_test.go

```diff
@@ -119,7 +119,7 @@ func TestSessionFinishRollsBackTxn(t *testing.T) {
 	defer log.Scope(t).Close(t)
 	aborter := NewTxnAborter()
 	defer aborter.Close(t)
-	params, _ := createTestServerParams()
+	params, _ := createTestServerParamsAllowTenants()
```

Comment:
This one is a problem with shared-process mode but not with external.

Reply:
Took care of this in: https://github.com/cockroachdb/cockroach/pull/140662

Reply:
Finally I decided to solve it in this PR itself since there is an existing handy API to give tenant specific connection string `s.ApplicationLayer().PGUrl()`